charts/sn-platform/templates/_helpers.tpl

```diff
@@ -157,4 +157,92 @@ jvmOptions:
   gcLoggingOptions:
   {{- toYaml . | nindent 2 }}
   {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{/*
+AntiAffinity Rules
+*/}}
+{{- define "pulsar.antiAffinityRules" }}
+{{- if .thisAffinity.affinity.customRules }}
+{{- if and .Values.affinity.anti_affinity .thisAffinity.anti_affinity }}
+podAntiAffinity:
+  {{ .thisAffinity.type }}:
+    {{ if eq .thisAffinity.type "requiredDuringSchedulingIgnoredDuringExecution" }}
+    - labelSelector:
+        matchExpressions:
+          - key: "app"
+            operator: In
+            values:
+              - "{{ template "pulsar.name" . }}"
+          - key: "release"
+            operator: In
+            values:
+              - {{ .Release.Name }}
+          - key: "component"
+            operator: In
+            values:
+              - {{ .Component }}
+      topologyKey: "kubernetes.io/hostname"
+      {{ if and .Values.affinity.zone_anti_affinity .thisAffinity.zone_anti_affinity }}
+  preferredDuringSchedulingIgnoredDuringExecution:
+    - weight: {{ .thisAffinity.zone_anti_affinity_weight | .Values.affinity.zone_anti_affinity_weight | default "100" }}
```

Comment:
here the .Values.affinity.zone_anti_affinity_weight will be treated as a function and will throw error, seems we can't use such way to choose value based on priority, I think it has to be either
`{{ .thisAffinity.zone_anti_affinity_weight | default .Values.affinity.zone_anti_affinity_weight }}
`or 
`{{ .thisAffinity.zone_anti_affinity_weight | default "100" }}`